### PR TITLE
hsmtool rune

### PR DESCRIPTION
**Pull request by rustyrussell**
This allows creation of a master rune *before the node starts*.  You will still need to create a rune once the node starts, if you want commando to work (as it is only activated once it has generated one), but creating a default rune will give the identical results to the tool anyway.

Requested-by: @ShahanaFarooqui

### CODE REVIEW

File: tools/hsmtool.c [hunk 1]
```diff
@@ -67,17 +70,25 @@ static bool ensure_hsm_secret_exists(int fd, const char *path)
...
+	if (!contents)
+		errx(EXITCODE_ERROR_HSM_FILE, "Reading hsm_secret");
 
-	fd = open(hsm_secret_path, O_RDONLY);
-	if (fd < 0)
-		errx(EXITCODE_ERROR_HSM_FILE, "Could not open hsm_secret");
-	if (!read_all(fd, hsm_secret, sizeof(*hsm_secret)))
-		errx(EXITCODE_ERROR_HSM_FILE, "Could not read hsm_secret");
-	close(fd);
+	/* grab_file always appends a NUL char for convenience */
```
Comment (vincenzopalazzo) [inline tools/hsmtool.c:80]:
```suggestion
	/* grab_file always appends a NULL char for convenience */
```

**Review by vincenzopalazzo — APPROVED** [review]
ACK https://github.com/ElementsProject/lightning/pull/6097/commits/6d4e942883cdad0dedd45c878b5355ac91fcff2b

**Review by ShahanaFarooqui — APPROVED** [review]
The tool works as expected but the cli hangs if commando is used with this master rune without calling commando-rune first. For posterity, can we add the message `You will still need to create a rune once the node starts, if you want commando to work (as it is only activated once it has generated one)` somewhere in this command usage?

File: tools/hsmtool.c (hunk 1 above)
Comment (rustyrussell) [reply tools/hsmtool.c:80]: No, confusingly "NUL" is the ASCII zero, NULL is the C pointer!

**Comment by rustyrussell** [comment]
OK, I added this to the documentation.  There's no really great place to add it in the tool, unfortunately: the usage message is all single lines, and the output is simply the rune: if we add a warning there it gets harder to parse.

**Comment by ShahanaFarooqui** [comment]
ACK [a914288](https://github.com/ElementsProject/lightning/commit/a9142888021fba6e6bde177b5a33b2ace5c77cb5)

**Pull request merged**